src/main/java/com/pinterest/secor/parser/MessageParser.java

```diff
@@ -44,4 +50,30 @@ public ParsedMessage parse(Message message) throws Exception {
     }
 
     public abstract String[] extractPartitions(Message payload) throws Exception;
+    
+    public Object getJsonFieldValue(JSONObject jsonObject) {
+        Object fieldValue = null;
+        String messageTimestampName = mConfig.getMessageTimestampName();
+        String messageTimestampSeparator = mConfig.getMessageTimestampNameSeparator();
+        if (messageTimestampSeparator != null) {
+            String[] fields = messageTimestampName.split(Pattern.quote(messageTimestampSeparator));
```

Comment:
Since you need to create a pattern object, this is probably better done in the constructor/initializer of the MessageParser, rather than inside of loop of message parsing.